pkg/middlewares/accesslog/logger.go

```diff
@@ -205,17 +208,32 @@ func (h *Handler) ServeHTTP(rw http.ResponseWriter, req *http.Request, next http
 		core[ClientUsername] = usernameIfPresent(reqWithDataTable.URL)
 	}
 
-	logDataTable.DownstreamResponse = crw.Header()
+	logDataTable.DownstreamResponse = downstreamResponse{
+		headers: deepCopy(crw.Header()),
```

Comment:
Couldn't you use `crw.Header().Clone()`? It does exactly what your `deepCopy` function do!